lib/rubygems/errors.rb

```diff
@@ -20,6 +20,41 @@ class LoadError < ::LoadError
     attr_accessor :requirement
   end
 
+  class MissingSpecError < Gem::LoadError
+    def initialize name, requirement
```

Comment:
@ka8725 [Are you sure?](https://github.com/rubygems/rubygems/blob/b326d3d40ada15d3e4434d9d6ad02dd1687c4608/lib/rubygems/errors.rb#L38).  I think new code is supposed to omit them.